### PR TITLE
Issue #64: Reset git repo to the initial state after report generation completed

### PR DESCRIPTION
**Pull request by HuGanghui**
Issue #64: Reset git repo to the initial state after report generation completed

### CODE REVIEW

File: src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java [hunk 1]
```diff
@@ -191,11 +195,29 @@ public void generatePatch(Set<String> commits) throws Exception {
...
+        final String headSha;
+        if (fileList != null && fileList[0] != null) {
+            final BufferedReader bufferedReader = new BufferedReader(new FileReader(fileList[0]));
+            headSha = bufferedReader.readLine();
+            fileList[0].delete();
+        }
+        else {
+            throw new IOException("Getting HEAD_SHA File failed!");
+        }
+        return headSha;
```
Comment (baratali) [inline src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java:220]: you can replace it with:
```java
Path path = new File(repoPath, "HEAD_SHA").toPath();
return Files.readAllLines(path).get(0);
```

Comment (HuGanghui) [reply src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java]: done

**Comment by rdiachenko** [comment]
@HuGanghui please rebase on latest master

**Comment by HuGanghui** [comment]
@rdiachenko rebased

**Pull request merged**